docs/messaging/kafka.md

```diff
@@ -44,7 +44,63 @@ If an intermediary broker is present, `service.name` and `peer.service` will not
 
 `messaging.client_id` SHOULD be set to the `client-id` of consumers, or to the `client.id` property of producers.
 
-## Examples
+## Metrics
+
+### Metric: `messaging.publish.duration`
+
+Measures time it takes to publish message or a batch of messages to Apache Kafka including all retries.
+
+This metric is [required][MetricRequired]
+
+This metric follows the common [messaging.publish.duration](./messaging-metrics.md#metric-messagingpublishduration) definition.
+
+**Notes:**
+
+- In addition to generic attributes defined for the metric, Apache Kafka instrumentation SHOULD report following attributes:
+  * `messaging.kafka.destination.partition`
+
+### Metric: `messaging.publish.messages`
+
+Measures the number of published messages.
+
+This metric is [required][MetricRequired]
+
+This metric follows the common [messaging.publish.messages](./messaging-metrics.md#metric-messagingpublishmessages) definition.
+
+**Notes:**
+
+- In addition to generic attributes defined for the metric, Apache Kafka instrumentation SHOULD report following attributes:
+  * `messaging.kafka.destination.partition`
+
+### Metric: `messaging.receive.duration`
+
+Measures duration of consumer client call that polls messages.
+
+This metric is [required][MetricRequired].
+
+This metric follows the common [messaging.receive.duration](./messaging-metrics.md#metric-messagingdeliverduration) definition.
+
+**Notes:**
+
+- In addition to generic attributes defined for the metric, Apache Kafka instrumentation SHOULD report following attributes:
+  * `messaging.kafka.destination.partition`
+  * `messaging.kafka.consumer.group`
+
+### Metric: `messaging.receive.messages`
+
+Measures the number of received messages.
+
+This metric is [required][MetricRequired].
+
+This metric follows the common [messaging.receive.messages](./messaging-metrics.md#metric-messagingreceivemessages) definition.
+
+**Notes:**
+
+- In addition to generic attributes defined for the metric, Apache Kafka instrumentation SHOULD report following attributes:
+  * `messaging.kafka.destination.partition`
+  * `messaging.kafka.consumer.group`
```

Comment:
Ideally, we'd be able to derive this information from YAML files. I think we don't have such a mechanism yet, but it would be necessary to support @lquerel's work going forward.

Reply:
yep, we have an issue for  it - https://github.com/open-telemetry/semantic-conventions/issues/592

Reply:
Resolving this thread based on SIG discussion - tooling is not a prereq for this PR and (probably) stability